src/scripts/main.js

```diff
@@ -1,3 +1,182 @@
 'use strict';
 
-// write your code here
+const gameScore = document.querySelector('.game-score');
+const button = document.querySelector('.button');
+
+const messageLose = document.querySelector('.message-lose');
+const messageWin = document.querySelector('.message-win');
+const messageStart = document.querySelector('.message-start');
+
+const fieldRow = document.querySelectorAll('.field-row');
+
+let score = 0;
+let board = initBoard();
+
+function initBoard() {
+  return [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
+};
+
+button.addEventListener('click', startGame);
+
+function gameReset() {
+  board = initBoard();
+  score = 0;
+  messageLose.classList = 'message message-lose hidden';
+  messageWin.classList = 'message message-win hidden';
+};
+
+function renderGameBoard() {
+  for (let row = 0; row < 4; row++) {
+    for (let column = 0; column < 4; column++) {
+      fieldRow[row].children[column].textContent
+      = board[row][column] ? board[row][column] : '';
+
+      fieldRow[row].children[column].className
+      = `field-cell field-cell--${board[row][column]}`;
+    }
+  }
+
+  gameScore.textContent = score;
+}
+
+function getRandomNumber() {
+  return Math.floor(Math.random() * 4);
+}
+
+function setRandomCell() {
+  const [x, y] = [
+    getRandomNumber(),
+    getRandomNumber(),
+  ];
+
+  if (!board[x][y]) {
+    board[x][y] = Math.random() >= 0.9 ? 4 : 2;
+    gameCheck();
+  } else {
+    setRandomCell();
+  }
+}
+
+function gameCheck() {
```

Comment:
```suggestion
function checkGame() {
```
The function's name should start with a verb
![image](https://github.com/mate-academy/js_2048_game/assets/96409563/a8e6cf1d-1477-49b2-a8e5-f91e5acf51f1)